src/praisonai-agents/praisonaiagents/tools/crowpay_tools.py

```
@@ -0,0 +1,235 @@
+        api_key: CrowPay API key (crow_sk_...)
+        approval_id: The approvalId from a 202 response
+
+    Returns:
+        Status dict. Terminal states: payload/sptToken present (approved),
+        status=denied, status=timeout, status=failed
+    """
+    try:
+        from urllib.request import urlopen, Request
+
+        url = f"{CROWPAY_BASE_URL}/authorize/status?id={approval_id}"
+        req = Request(url, headers={"X-API-Key": api_key})
```

Ruff (0.15.2)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)